backend/tracim_backend/lib/core/workspace.py

```diff
@@ -261,6 +261,32 @@ def execute_created_workspace_actions(self, workspace: Workspace) -> None:
                 except AgendaServerConnectionError as exc:
                     logger.error(self, 'Cannot connect to agenda server')
                     logger.exception(self, exc)
+                except Exception as exc:
+                    logger.error(self, 'Something goes wrong during agenda create/update')
+                    logger.exception(self, exc)
+
+    def execute_update_workspace_actions(self, workspace: Workspace) -> None:
+        from tracim_backend.lib.agenda.agenda import AgendaApi
+        if self._config.CALDAV_ENABLED:
+            agenda_api = AgendaApi(
```

Comment:
do not instantiate an object if it may be useless. agenda_api should be instanciated in the `if workspace.agenda_enabled:`

Reply:
done.